package/network/services/wireguard/files/wireguard_watchdog

```diff
@@ -0,0 +1,52 @@
+#!/bin/sh
+# SPDX-License-Identifier: GPL-2.0
+#
+# Copyright (C) 2018 Aleksandr V. Piskunov <aleksandr.v.piskunov@gmail.com>.
+# Copyright (C) 2015-2018 Jason A. Donenfeld <Jason@zx2c4.com>. All Rights Reserved.
+#
+# This watchdog script tries to reresolve and reconnect to inactive wireguard peers.
+# Use it for peers with a frequently changing dynamic IP. 
+# persistent_keepalive must be set, recommended value is 25 seconds.
+#
+# Run this script from cron every minute:
+# echo '* * * * * /usr/bin/wireguard_watchdog' >> /etc/crontabs/root
+
+
+. /lib/functions.sh
+
+check_peer_activity() {
+  local cfg=$1
+  local iface=$2
+  local public_key
+  local endpoint_host
+  local endpoint_port
+  local persistent_keepalive
+  local last_handshake
+  local idle_seconds
+
+  config_get public_key "${cfg}" "public_key"
+  config_get endpoint_host "${cfg}" "endpoint_host"
+  config_get endpoint_port "${cfg}" "endpoint_port"
+  config_get persistent_keepalive "${cfg}" "persistent_keepalive"
+
+  # only process peers with endpoints and keepalive set
+  [ -z ${endpoint_host} ] && return 0;
+  [ -z ${persistent_keepalive} ] && return 0;
+
+  # resolve endpoint and reconnect if not responding for too long
+  last_handshake=`wg show ${iface} latest-handshakes | grep ${public_key} | awk '{print $2}'`
+  [ -z ${last_handshake} ] && return 0;
+  idle_seconds=$((`date +%s`-${last_handshake}))
+  [ ${idle_seconds} -lt 150 ] && return 0;
+  logger -t "wireguard_monitor" "${iface} endpoint ${endpoint_host}:${endpoint_port} is not responding for ${idle_seconds} seconds, trying to reconnect"
```

Comment:
"reconnect" is probably the wrong word here. Maybe, instead, "re-resolve host"?

Reply:
changed wording to "re-resolve" in comments, log message and package description